packages/bruno-app/src/components/Sidebar/Collections/Collection/CollectionItem/GenerateCodeItem/CodeView/index.js

```diff
@@ -20,10 +21,16 @@ const CodeView = ({ language, item }) => {
     item.uid
   );
 
-  const headers = [...(collection?.root?.request?.headers || []), ...(requestHeaders || [])];
+  const collectionRootAuth = collection?.root?.request?.auth;
+  const requestAuth = item.draft ? get(item, 'draft.request.auth') : get(item, 'request.auth');
 
-  let snippet = '';
+  const headers = [
+    ...(collection?.root?.request?.headers || []),
+    ...(requestHeaders || []),
+    ...getAuthHeaders(collectionRootAuth, requestAuth)
+  ];
```

Comment:
The list of headers is translated to a map. Previously mapped items may be overridden. If headers with the key "Authorization" are declared at the request level, they will be overridden by collection-level auth.

Reply:
thanks i will change this!